lua/wire/gates/string.lua

```diff
@@ -131,13 +133,23 @@ GateActions["string_find"] = {
 	end
 }
 
-
 GateActions["string_concat"] = {
 	name = "Concatenate",
 	inputs = { "A" , "B" , "C" , "D" , "E" , "F" , "G" , "H" },
 	inputtypes = { "STRING" , "STRING" , "STRING" , "STRING" , "STRING" , "STRING" , "STRING" , "STRING" },
 	outputtypes = { "STRING" },
 	output = function(gate, A, B, C, D, E, F, G, H)
+		if  (A and #A or 0) 
+		  + (B and #B or 0) 
+		  + (C and #C or 0) 
+		  + (D and #D or 0) 
+		  + (E and #E or 0) 
+		  + (F and #F or 0) 
+		  + (G and #G or 0) 
+		  + (H and #H or 0)  > MAX_LEN   
+		then
+			return false
```

Comment:
Yeah, if the gate is supposed to output a string then it should actually return a string (unless there's some special handling of incorrectly-typed values that presents errors in a graceful way).